floyd/cli/data.py

```diff
@@ -141,6 +141,66 @@ def clone(id):
                               delete_after_untar=True)
 
 
+@click.command()
+@click.argument('id', nargs=1)
+def listfiles(id):
+    """
+    List files in the given dataset
+    """
+
+    data_source = DataClient().get(normalize_data_name(id, use_data_config=False))
+    if id and not data_source:
+        # Try with the raw ID
+        data_source = DataClient().get(id)
+
+    if not data_source:
+        if 'output' in id:
+            floyd_logger.info("Note: You cannot clone the output of a running job. You need to wait for it to finish.")
+        sys.exit()
+
+    # Depth-first search
+    dirs = ['']
+    paths = []
+    while dirs:
+        dir = dirs.pop()
+        url = "/resources/{}/{}?content=true".format(data_source.resource_id, dir)
```

Comment:
there is also a `skipped_files` entry which could be useful to notify the user that they are not getting the full list.

Reply:
Under what conditions is `skipped_files` set to `True`?

Reply:
it's actual a number that tells you how many files are skipped.

Reply:
Ah, OK. What can make files be skipped sometimes?

Reply:
we have a limit of how many files within a directory to index, right now it's set to 250, but subject to change in the future.

Reply:
Oh, gosh, right. Yeah, I'll add a message telling users about that then.